include/fiction/algorithms/physical_design/post_layout_optimization.hpp

```diff
@@ -1035,6 +531,567 @@ class post_layout_optimization_impl
      * Statistics about the post-layout optimization process.
      */
     post_layout_optimization_stats& pst;
+    /**
+     * Maximum number of relocations to try for each gate.
+     */
+    std::optional<uint64_t> max_gate_relocations = std::nullopt;
+    /**
+     * Maximum coordinate of all gates that are not POs.
+     */
+    tile<Lyt> max_non_po{0, 0};
+    /**
+     * Timeout limit reached.
+     */
+    bool timeout_limit_reached = false;
+    /**
+     * Start time.
+     */
+    std::chrono::time_point<std::chrono::high_resolution_clock> start;
```

Comment:
It is a best practice to declare members in the same order they are constructed in, putting all the ones instantiated by the constructor first. In this case, you would move `start` over `max_gate_relocations`.